src/main/java/de/tum/in/www1/artemis/config/migration/entries/MigrationEntry20220619_123000.java

```diff
@@ -0,0 +1,145 @@
+                latestUnprocessedResultDate = (Stream.concat(unprocessedResults.stream().map(Result::getCompletionDate), Stream.of(latestUnprocessedResultDate)))
+                        .filter(Objects::nonNull).max(Comparator.naturalOrder()).orElse(null);
+            }
+        }
+
+        if (numberOfUnprocessedResults != 0) {
+            LOGGER.error(
+                    "Found {} results of programming exercises that could not be processed! These results will no longer show a correct resultString! The actual score is unaffected by this.",
+                    numberOfUnprocessedResults);
+            if (latestUnprocessedResultDate != null) {
+                LOGGER.error("The latest of these unprocessed results is from {}.", latestUnprocessedResultDate);
+            }
+        }
+    }
+
+    /*
+     * Takes a result string such as "21 of 42 passed, 42 issues" and splits it into its different components: The number of tests, the number of passed tests and the number of
+     * issues.
+     * @param results Batch of results that should get processed
+     */
+    private List<Result> processProgrammingResults(List<Result> results, Field resultString) {
+        List<Result> unsuccessfulResults = results.stream().filter(result -> {
+            String[] resultStringParts = getResultString(result, resultString).split(", ");
+            boolean unsuccessful = true;
+            for (String resultStringPart : resultStringParts) {
+                // Matches e.g. "21 of 42 passed"
+                if (resultStringPart.matches(".*of.*passed.*")) {
+                    String[] testCaseParts = resultStringPart.split(" ");
+                    int passedTestCasesAmount = Integer.parseInt(testCaseParts[0]);
+                    int testCasesAmount = Integer.parseInt(testCaseParts[2]);
+
+                    result.setPassedTestCaseCount(passedTestCasesAmount);
+                    result.setTestCaseCount(testCasesAmount);
+
+                    // If we found the test cases, we successfully migrated that result
+                    unsuccessful = false;
+                }
+                // Matches e.g. "9 issues"
+                else if (resultStringPart.contains("issue")) {
+                    String[] issueParts = resultStringPart.split(" ");
+                    int codeIssueCount = Integer.parseInt(issueParts[0]);
+
+                    result.setCodeIssueCount(codeIssueCount);
+                }
+            }
+            return unsuccessful;
+        }).toList();
+
+        resultRepository.saveAll(results);
+
+        return unsuccessfulResults;
+    }
+
+    private String getResultString(Result result, Field resultString) {
+        try {
+            return (String) resultString.get(result);
+        }
+        catch (IllegalAccessException e) {
+            throw new RuntimeException("Could not access the field \"resultString\" of the class Result during the migration!");
```

Comment:
Maybe also add to the message here that an intermediary upgrade to Artemis version 5.8.5 before upgrading to this version might solve the issue.